work_queue/src/work_queue.c

```diff
@@ -827,6 +850,72 @@ static int get_transfer_wait_time(struct work_queue *q, struct work_queue_worker
 	return timeout;
 }
 
+static void remove_factory(struct work_queue_factory_info *f)
+{
+	free(f->name);
+	free(f);
+}
+
+static void update_factory(struct work_queue *q, struct jx *j)
+{
+	const char *name = jx_lookup_string(j, "factory_name");
+	struct work_queue_factory_info *f;
+
+	if ( !(f = hash_table_lookup(q->factory_table, name)) ) {
+		// Create a new factory
+		debug(D_WQ, "new factory %s detected", name);
+		f = malloc(sizeof(*f));
+		if (!f) {
+			debug(D_NOTICE, "Cannot allocate memory for factory %s.", name);
+			return;
+		}
+		memset(f, 0, sizeof(*f));
+	}
+
+	f->name        = xxstrdup(name);
+	f->num_workers = 0;
```

Comment:
num_workers should not be reset here. Rather, we should set the max_workers from the value we got from the catalog for that factory.

Reply:
There's something wrong with the whole logic here. Now it's fixed.